java/examples/datasets/JavaDatasetExample.sh.in

```diff
@@ -36,8 +37,8 @@ nerrors=0
 
 # where the libs exist
 HDFLIB_HOME="$top_srcdir/java/lib"
-BLDLIBDIR="$top_builddir/hdf5/lib"
```

Comment:
Changing BLDLIBDIR isn't really necessary;  lines 39 - 41 could stay as they were.  The rpath change added in lines 117- 123 is what's needed to make the examples tests pass on macOS.

Reply:
I'll approve this conditional on this being addressed.